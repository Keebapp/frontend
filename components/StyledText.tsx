import { Text, TextProps } from './Themed';

export function RegularText(props: TextProps) {
	return <Text {...props} style={[props.style, { fontFamily: 'SourceSansProRegular' }]} />;
}

export function BoldText(props: TextProps) {
	return <Text {...props} style={[props.style, { fontFamily: 'SourceSansProBold' }]} />;
}

export function ItalicText(props: TextProps) {
	return <Text {...props} style={[props.style, { fontFamily: 'SourceSansProItalic' }]} />;
}

export function ItalicBoldText(props: TextProps) {
	return <Text {...props} style={[props.style, { fontFamily: 'SourceSansProBoldItalic' }]} />;
}
