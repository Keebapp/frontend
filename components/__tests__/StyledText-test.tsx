import * as React from 'react';
import renderer from 'react-test-renderer';

import { RegularText } from '../StyledText';

it('renders correctly', () => {
	const tree = renderer.create(<RegularText>Snapshot test!</RegularText>).toJSON();

	expect(tree).toMatchSnapshot();
});
