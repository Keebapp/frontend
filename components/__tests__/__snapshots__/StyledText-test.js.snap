// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#fff",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "SourceSansProRegular",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
